app/src/main/java/com/android/sample/model/auth/SignInRepository.kt

```diff
@@ -1,19 +1,11 @@
 package com.android.sample.model.auth
 
-import com.android.sample.ui.navigation.NavigationActions
 import com.google.firebase.auth.AuthResult
 
 interface SignInRepository {
   suspend fun signInWithEmail(email: String, password: String): AuthResult
 
   suspend fun signInWithGoogle(idToken: String): AuthResult
 
-  fun checkUserProfile(
-      uid: String?,
-      navigationActions: NavigationActions,
-      onAuthSuccess: () -> Unit,
-      onAuthError: (String) -> Unit
-  )
-
```

Comment:
Yes, a function implemented in the viewmodel should not appear in the repository interface